### PR TITLE
Adds optional Wheres attribute on class and method level

### PR DESCRIPTION
**Pull request by obuchmann-pc-web**
Regarding the discussion #29 I wrote down my idea of implementing where.

I added a new Attribute `Where` and made it available for class and method usage.

Usage:
```php

#[Where('my-where', '[0-9]+')]
class MyController
{
    #[Get('my-get-route/{my-where}')]
    public function myGetMethod()
    {
    }

    #[Post('my-post-route/{my-where}/{my-alpha-numeric}')]
    #[WhereAlpha('alpha')]
    #[WhereAlphaNumeric('alpha-numeric')]
    #[WhereNumber('number')]
    #[WhereUuid('uuid')]
    public function myPostMethod()
    {
    }
}
``` 

~~We can improve this by implementing the [shorthands laravel implemented](https://laravel.com/docs/8.x/routing#parameters-regular-expression-constraints) either as own Attribute or as parameter for the Wheres attribute.
I'll leave this open for discussion.~~

Todo:
- [x]  Write docs
- [x] Implement shorthands?

### CODE REVIEW

File: src/Attributes/Wheres.php [hunk 1]
```diff
@@ -0,0 +1,14 @@
+<?php
+
+namespace Spatie\RouteAttributes\Attributes;
+
+use Attribute;
+
+#[Attribute(Attribute::TARGET_METHOD | Attribute::TARGET_CLASS)]
+class Wheres implements RouteAttribute
```
Comment (freekmurze) [inline src/Attributes/Wheres.php:8]: Rename this to `Where`, feels much more natural to me.

**Review by freekmurze — CHANGES_REQUESTED** [review]
Thanks for your work on this.

File: src/Attributes/Wheres.php (hunk 1 above)
Comment (obuchmann-pc-web) [reply src/Attributes/Wheres.php:8]: I named it Where first but due to it takes multiple arguments i renamed it to wheres.
There is also the possibility to implement it with multiple attributes.

e.g.:

```
    #[Get('my-where-method/{param}/{param2}')]
    #[Where('param2', '[a-zA-Z]+'], Where('param', '[0-9]+')]
    public function myWhereMethod()
    {

    }
```

This leads to more complicated resolving strategy (collecting multiple attributes) but feels natural when using.

Comment (freekmurze) [reply src/Attributes/Wheres.php:8]: Let's optimise for that natural feeling when using it.

Comment (obuchmann-pc-web) [reply src/Attributes/Wheres.php]: I implemented the single Where version and some convenience attributes. I updated the docs.

**Pull request merged**